packages/material/src/complex/MaterialObjectRenderer.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+    ControlElement,
+    ControlProps,
+    isObjectControl,
+    mapStateToControlProps,
+    RankedTester,
+    rankWith,
+    resolveSchema,
+    UISchemaElement
+} from '@jsonforms/core';
+import { connectToJsonForms } from '@jsonforms/react';
+import { MaterialLayoutRenderer } from '../util/layout';
+import { Card, CardContent, CardHeader } from '@material-ui/core';
+
+const createControl = (schema, path: string): UISchemaElement[] => {
+    const obj =  resolveSchema(schema, path);
+    return Object.keys(obj).map(key => {
+        return {
+            label: key,
+            type: 'Control',
+            scope: `${path}/${key}`
+        };
+    });
+};
+
+class MaterialObjectRenderer extends React.Component<ControlProps, any> {
+
+    render() {
+        const {
+            uischema,
+            schema,
+            visible,
+           } = this.props;
+
+        const controlElement = uischema as ControlElement;
+        const scope = controlElement.scope;
+        const elements: UISchemaElement[] = createControl(schema, (`${scope || '#'}/properties`));
```

Comment:
we should retrieve or at least generate the elements using one of the existing functions instead of doing this with custom code

Reply:
I've exposed `createControlElement` from the UI schema generator. Does this address your concern?